From a6cfc7860a7a1d828ec19190c4043b9f9aed082f Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Tue, 30 Apr 2013 17:19:18 +0200
Subject: Add Stream Flags key to setParameters

BZ: 70126
[CLEAN]

Audio Route Manager needs to know the type of stream to take the
appropriate routing decision (eg direct Output, MP3 Offload).
It must be define as key and filled by the policy.

Category: device enablement
Domain: AudioComms-Common
Origin: internal
Upstream-Candidate: yes
Change-Id: I43a568c0333ead6c9acf15ca66563444109e9c57
Orig-Change-Id: Ib0f538092d997254bab1af147eeca90647a88e5d
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 audio/AudioPolicyManagerBase.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index bb19830..31a4417 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2759,6 +2759,7 @@ uint32_t AudioPolicyManagerBase::setOutputDevice(audio_io_handle_t output,
         force (%d) delayMs (%d) on Output (%d)", prevDevice, device, force, delayMs, output);
     // do the routing
     param.addInt(String8(AudioParameter::keyRouting), (int)device);
+    param.addInt(String8(AudioParameter::keyStreamFlags), (int)outputDesc->mFlags);
     mpClientInterface->setParameters(output, param.toString(), delayMs);
 
     // update stream volumes according to new device
-- 
2.5.0

